File: sake/Models/Item.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Divider.Models;

public class Item : BaseEntity
{
  public string Name { get; set; } = string.Empty;
  public int? Quantity { get; set; }
  public float Price { get; set; }

  [ForeignKey("ContactId")]
  public int ContactId { get; set; }
}

File: sake/models/Item.ts
import { ObjectId, WithId } from "mongodb";

export interface Item extends WithId<Document> {
  name: string;
  price: number;
  // optional param to use for calcs (not for storage)
  assignedContactsCount?: number; 
}

export interface ItemAssignment {
  contactId: ObjectId;
  itemId: ObjectId;
}
